package.json

```diff
@@ -71,6 +71,35 @@
           "type": "boolean",
           "default": false,
           "description": "Show inlay hints even there are spread operators in the array (will use ? for the items after spread)"
+        },
+        "arrayIndexInlay.depthIndicator": {
+          "type": "string",
+          "default": "Simple",
```

Comment:
Maybe default to `none`? Or can you explain a bit more why this could be useful in general?

Reply:
The # prefix before each number seemed like it could be used more effectively. This gives a way to visually see depth in an array regardless of indentation of code / counting spaces yourself. In heavily nested data I think it can provide a lot of value.

Reply:
My concern is that sometimes it might give an illusion of the actually array depth, as from a parser, we can't really now the actual depth of an array, plus it would have cases like combining with nested objects/classes/arguements etc.

So what I feel this feature is more like "differentiate different level/arrays"? In that case, what about assigning them with different colors?

Reply:
I hadn't fully thought it through I guess, with nested objects etc. Makes sense to treat it similar to bracket colorisation.

![image](https://github.com/user-attachments/assets/42dbe9ff-502d-4d6e-8560-7b7436f0d8e4)

This is using the color variables for bracket colorisation. I'm not sure if these could become out of sync, in which case it would defintely be confusing